shared/actions/fs/index.js

```diff
@@ -935,6 +935,15 @@ const onFSOverallSyncSyncStatusChanged = (state, action) =>
     progress: Constants.makeSyncingFoldersProgress(action.payload.params.status.prefetchProgress),
   })
 
+let prevOutOfSpace = false
+const updateMenubarIconOnStuckSync = (state, action) => {
+  const outOfSpace = action.payload.params.status.outOfSyncSpace
+  if (outOfSpace !== prevOutOfSpace) {
+    prevOutOfSpace = outOfSpace
+    return NotificationsGen.createBadgeApp({key: 'outOfSpace', on: outOfSpace})
+  }
+}
```

Comment:
It'd be nice if we could do this sort of deduplication in Go side.

Reply:
The Go side is sending us overall status notifications once per second, which include the current space status and the current sync status. Currently the space status is just a boolean value for whether it's out of space or not - it doesn't seem worth it to me to complicate that to (out of space, not out of space, unchanged).